### PR TITLE
phabricator: Use an exponential backoff sleep time

### PR DESCRIPTION
**Pull request by La0**
Using tenacity would require a large refactoring, as the method is not retried in a direct loop, but through the usage of the libmozvent queues.

Fixes https://github.com/mozilla/code-review/issues/328

### CODE REVIEW

File: libmozevent/phabricator.py [hunk 1]
```diff
@@ -91,8 +91,9 @@ def update_state(self, build):
             return
 
         # Check this build has been awaited between tries
+        exp_backoff = (2 ** (self.retries - retries_left + 1) - 1) * self.sleep
```
Comment (marco-c) [inline libmozevent/phabricator.py]: `self.retries - retries_left` won't work, as self.retries is a dict, no?

**Comment by marco-c** [comment]
With this, in total we will wait 10 * 2^6, so 640 seconds. We used to wait 50 seconds. Maybe we should wait up to 320, so use 4 retries?

File: libmozevent/phabricator.py (hunk 1 above)
Comment (La0) [reply libmozevent/phabricator.py]: Weird that the unit test on code review did not pick that up

Comment (La0) [reply libmozevent/phabricator.py]: I'm adding a local unit test to trigger it

File: libmozevent/phabricator.py [hunk 2]
```diff
@@ -91,8 +92,9 @@ def update_state(self, build):
             return
 
         # Check this build has been awaited between tries
+        exp_backoff = (2 ** (self.max_retries - retries_left + 1) - 1) * self.sleep
```
Comment (marco-c) [inline libmozevent/phabricator.py]: Nit: you could do `(2 ** (self.max_retries - retries_left)) * self.sleep` and set max_retries to 5 to have more or less the same effect but simpler code.

**Pull request merged**